### PR TITLE
Handle unknown statement while parsing profile. Fixes #6931

### LINKED ISSUE #6931: [feature] Confusing error message when typo in profile
I had a typo in my profile file:

```
includes(cmake_3_16)

[build_requires]
doxygen/1.8.15@company/stable
```

Where it should have been `include` instead of `includes`. 
When running Conan, I got the following error message:

```
ERROR: Error reading 'documentation' profile: Error parsing the profile text file: not enough values to unpack (expected 2, got 1)
```

At least I knew that something is wrong with the profile, but the error message was a bit misleading
A more helpful message could have been:

```
ERROR: Error reading 'documentation' profile: Error parsing the profile text file: includes is not a valid keyword
```
or something similar.

### Environment Details (include every applicable attribute)
  * Conan version: 1.24.1

### PR DESCRIPTION
**Pull request by iblancasa**
Changelog: Bugfix: Provide a more descriptive error when an unknown statement is added to a profile
Docs: omit

Closes: #6931

Tested with the example provided in the [examples repository](https://github.com/conan-io/examples/tree/master/features/emscripten)

I replaced the first line:
```
include(default)
```

By:
```
includes(default)
```

When running `conan` before this change, the command output is:
```sh
✗ ./build.sh
+ conan remove 'conan-hello-emscripten/*' -f
WARN: No package recipe matches 'conan-hello-emscripten/*'
+ conan create . conan/testing -pr emscripten.profile --build missing
ERROR: Error reading 'emscripten.profile' profile: Error parsing the profile text file: not enough values to unpack (expected 2, got 1)
```

After this change:
```sh
✗ ./build.sh
+ conan remove 'conan-hello-emscripten/*' -f
WARN: No package recipe matches 'conan-hello-emscripten/*'
+ conan create . conan/testing -pr emscripten.profile --build missing
ERROR: Error reading 'emscripten.profile' profile: Error while parsing line 0: 'includes(default)'
```

### CODE REVIEW

File: conans/client/profile_loader.py [hunk 1]
```diff
@@ -38,7 +38,10 @@ def __init__(self, text):
                 include = include[:-1]
                 self.includes.append(include)
             else:
-                name, value = line.split("=", 1)
+                try:
+                    name, value = line.split("=", 1)
+                except ValueError as error:
+                    raise ConanParsingError("Error while parsing line %i: '%s'" % (counter, line))
```
Comment (memsharded) [inline conans/client/profile_loader.py]: Let's throw a generic ``ConanException``. New exceptions will be introduced from now on, only if it is going to be captured and processed somewhere else. Let's keep it simple.

**Review by memsharded — COMMENTED** [review]
Fix looks good. Please check the comment about the exception.

It would be great to include a test that covers this case. Do you think you could manage to add it? ``profile_loader_test.py`` seems a good place. Ask for help if need it, we could contribute the test if you prefer.

**Comment by memsharded** [comment]
I hadn't check the CI, please check: https://conan-ci.jfrog.info/blue/organizations/jenkins/ConanTestSuite/detail/PR-7577/1/pipeline

One of the test was producing a different output, the test can be changed (and also adding a new one that valides the ``includes()`` error too)

**Comment by iblancasa** [comment]
Ok, I added the requested changes except the tests. I'm trying to figure out what test I broke because, when I run `tox`, I get:
```sh
FAILED (SKIP=250, errors=30, failures=2)
```

Even if I run those tests from `develop`. I'm running the tests in a Fedroa 31 machine. Can be related? Do I need a specific environment? If yes, do you think it can make sense to create a Dockerfile to help developers to run tests?

**Comment by memsharded** [comment]
> Even if I run those tests from develop. I'm running the tests in a Fedroa 31 machine. Can be related? Do I need a specific environment? If yes, do you think it can make sense to create a Dockerfile to help developers to run tests?

Yes, setting up the whole environment might be a bit complicated, and a docker image will only partially help, because there might be easily tests failing in other platforms, other python versions, etc. Typically having a look at our CI and running those failing tests manually (with ``nosetests conans.test.functional...``) is good enough for development, this is what I do many times, when not in my main OS.

In your case: https://conan-ci.jfrog.info/blue/organizations/jenkins/ConanTestSuite/detail/PR-7577/2/pipeline, so probably running ``nosetests conans.test.functional.command.install.install_test.InstallTest`` is good enough.

**Review by memsharded — APPROVED** [review]
Perfect, thanks!

**Pull request merged**